js/scripts.js

```diff
@@ -90,43 +92,73 @@ document.addEventListener("DOMContentLoaded", function() {
         }
     }
 
-    // Function to filter persons based on category
+    const selectedCategory = document.getElementById('category');
+    selectedCategory.addEventListener('change', (event) => {
+        const category = event.target.value;
+        filterCategory(category);
+    });
+
     function filterCategory(category) {
-        let persons = document.getElementById('persons');
-        let person = persons.getElementsByClassName('card');
+        const cards = document.querySelectorAll('#persons .card');
 
-        for (let i = 0; i < person.length; i++) {
-            if (category === 'All' || person[i].getAttribute('data-category') === category) {
-                person[i].style.display = '';
-            } else {
-                person[i].style.display = 'none';
+        cards.forEach((card) => {
+            if (card.dataset.category.toLowerCase() !== category) {
+                card.style.display = 'none';
             }
-        }
+        })
+    }
+
+    // Filter by detaining police stations
+
+    const selectedPoliceStation = document.getElementById('police-station');
+    selectedPoliceStation.addEventListener('change', (event) => {
```

Comment:
element with id police-station doesn't exist

Reply:
that should be detained_at as per this https://github.com/wkambale/missingpersons/pull/45/files#diff-0eb547304658805aad788d320f10bf1f292797b5e6d745a3bf617584da017051R78 🤦🏽‍♀️